File: java/src/main/java/gov/doc/ntia/sigmf/ext/global/location/Datum.java
package gov.doc.ntia.sigmf.ext.global.location;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.io.Serializable;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class Datum implements Serializable {

    private static final long serialVersionUID = 1L;

    @JsonProperty(value="id", required = false)
    protected String id;

    @JsonProperty(value="description", required = false)
    protected String description;

    @JsonProperty(value= "spheroid", required = false)
    protected Spheroid spheroid;


    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Spheroid getSpheroid() {
        return spheroid;
    }

    public void setSpheroid(Spheroid spheroid) {
        this.spheroid = spheroid;
    }



}

File: Csharp/sigmf/sigmf/ext/global/location/Datum.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace sigmf
{
    public class Datum
    {
        [JsonProperty("id")]
        protected string id;

        [JsonProperty("description")]
        protected string description;

        [JsonProperty("spheroid")]
        protected Spheroid spheroid;

        public string Id { get => id; set => id = value; }
        public string Description { get => description; set => description = value; }
        public Spheroid Spheroid { get => spheroid; set => spheroid = value; }
    }
}
